dragonfly/implementations/at_operator.py

```diff
@@ -0,0 +1,472 @@
+from __future__ import absolute_import
+
+try:
+    import dateutil, funcsigs
+    # for google
+    import googleapiclient.discovery
+    from google.oauth2.credentials import Credentials
+except ImportError:
+    pass
+
+import datetime, logging, os, time, json
+# for slack
+import slackclient
+
+from dripline.core import Endpoint, fancy_doc
+from .subprocess_mixin import *
+
+__all__ = []
+__all__.append('AtOperator')
+
+logger = logging.getLogger(__name__)
+
+@fancy_doc
+class AtOperator(SlowSubprocessMixin, Endpoint):
+    
+    def __init__(self, 
+                 monitor_channel_name = 'slack_test',
+                 update_interval = {"hours":12},
+                 authentication_path = ".project8_authentications.json",
+                 **kwargs):
+        '''
+        monitor_channel_name: the name of Slack monitor channel. 
+        update_interval     : the time interval between regular checks and updates.
+        authentication_path : the absolute path of the file that stores the authentication info.
+        '''
+        imports = ['dateutil', 'funcsigs', 'googleapiclient', 'Credentials']
+        for import_module in imports:
+            if not import_module in globals():
+                raise ImportError(import_module + ' not found, required for AtOperator class.')
+
+        this_home = os.path.expanduser('~')
+        self.authentication_path = os.path.join(this_home, authentication_path)
+        try:
+            json.loads(open(self.authentication_path).read())
```

Comment:
curiosity: is there any functional difference between this and `json.load(open(self.authentication_path))`?

Reply:
I think both of them would work in this case...?